components/camera/replaypcd/replaypcd_test.go

```diff
@@ -424,3 +427,43 @@ func TestUnimplementedFunctions(t *testing.T) {
 
 	test.That(t, serverClose(), test.ShouldBeNil)
 }
+
+// TestNextPointCloudTimestamps tests that calls to NextPointCloud on the replay camera will inject
+// the time received and time requested metadata into the gRPC response header.
+func TestNextPointCloudTimestamps(t *testing.T) {
+	// Construct replay camera.
+	ctx := context.Background()
+	cfg := &Config{Source: "source"}
+	replayCamera, serverClose, err := createNewReplayPCDCamera(ctx, t, cfg, true)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, replayCamera, test.ShouldNotBeNil)
+
+	// Repeatedly call NextPointCloud, checking for timestamps in the gRPC header.
+	for i := 0; i < numPCDFiles; i++ {
+		serverStream := testutils.NewServerTransportStream()
```

Comment:
[q] Does the metadata context need to be created every time? If not we could move lines 443-444 to 435 and call:

`ctx = grpc.NewContextWithServerTransportStream(context.Background(), serverStream)`